### PR TITLE
feat: add support for saving / restoring rtx cache

### LINKED ISSUE #55: Integrate Caching
Would be great if caching of installed plugins was handled inside the action.  Maybe by passing a boolean input.  It's probably not necessary to cache the the executable since that's a quick download but the plugins would really speed up execution.

### PR DESCRIPTION
**Pull request by HipsterBrown**
Fixes #55 

I based the architecture and workflow on the [setup-node action](https://github.com/actions/setup-node/tree/main), including the log messages. I updated some fields in the package.json to more accurately reflect the name and info about this action; let me know if that is too much for one PR and I can revert those changes. 

At the moment, this is a fairly naive cache by including everything under the rtx directory (default `~/.local/share/rtx`) including the rtx binary itself. The cache key _should_ prevent cross-arch-platform incompatibility to be used, and this implementation doesn't perform any checks to skip installing the binary. In a different PR, I wonder if the [`@actions/tool-cache`](https://github.com/actions/toolkit/tree/main/packages/tool-cache) package could be used to download or restore the rtx binary, similar to how [setup-node does](https://github.com/actions/setup-node/blob/9d255ef2450ad5239d2bd23a57b08b3a7c2918f3/src/distributions/base-distribution.ts#L1).

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -1,17 +1,40 @@
...
   await setupRTX()
   await exec.exec('rtx', ['--version'])
-  await setToolVersions()
   await exec.exec('rtx', ['install'])
   await setPaths()
 }
 
+async function restoreRTXCache(): Promise<void> {
+  const cachePath = rtxDir()
+  const fileHash = await glob.hashFiles('**/.tool-versions')
```
[the remark this answers is not in the document]
Comment (HipsterBrown) [reply src/main.ts:21]: Will that ever be an input for this action, like `tool_versions` is?

[the remark this answers is not in the document]
Comment (jdxcode) [reply src/main.ts:21]: yes https://github.com/jdxcode/rtx-action/issues/47

File: dist/cache-save/index.js
Comment (jdxcode) [inline dist/cache-save/index.js:1]: any idea what this is? it's a massive file

Comment (HipsterBrown) [reply dist/cache-save/index.js:1]: This is the [ncc](https://npm.im/@vercel/ncc) build output for the `lib/cache-save.js` (after tsc for `src/cache-save.ts`), similar to the `dist/index.js` output for the `lib/main.js` (after tsc for `src/main.ts`). This is used by the `post` hook for the `action.yml` when the workflow is cleaning up.

**Pull request merged**